plugins/commands/create.js

```diff
@@ -1,10 +1,73 @@
+/* eslint no-underscore-dangle: ["error", { "allow": ["_data", "_shot"] }] */
+
 'use strict';
 
 const boom = require('boom');
 const schema = require('screwdriver-data-schema');
 const validator = require('screwdriver-command-validator');
 const hoek = require('hoek');
 const urlLib = require('url');
+const req = require('request');
+const VERSION_REGEX = schema.config.regex.VERSION;
+const DEFAULT_BYTES = 1024 * 1024 * 1024; // 1GB
+
+/**
+ * Publish binary command
+ * @method binaryCommandPublish
+ * @param  {CommandFactory} commandFactory      commandFactory
+ * @param  {Object}         config              Command config
+ * @param  {Binary}         binary              Binary published to the store
+ * @param  {String}         storeUrl            URL to the store
+ * @param  {String}         authToken           Bearer Token to be passed to the store
+ * @return {Promise}
+ */
+function binaryCommandPublish(commandFactory, config, binary, storeUrl, authToken) {
+    const [, major, minor] = VERSION_REGEX.exec(config.version);
+    const searchVersion = minor ? `${major}${minor}` : major;
+    let publishVersion;
+
+    return commandFactory.getCommand(`${config.namespace}/${config.name}@${searchVersion}`)
+        .then((latest) => {
+            if (!latest) {
+                publishVersion = minor ? `${major}${minor}.0` : `${major}.0.0`;
```

Comment:
I'm wondering why we need to do the duplicate work of this https://github.com/screwdriver-cd/models/blob/master/lib/commandFactory.js#L66-L78 here?

Can we call `return commandFactory.create(config)` (line 68) first, and that will return a command, and we can get `command.version`?

Reply:
Actually, we call `return commandFactory.create(config)` (line 68) first, and that will return a command, and we can get `command.version`. However, this will cause the meta to be saved and the binary files not to be saved if an error occurs while saving to the store.

Reply:
Okay. So you want the command to be created **ONLY AFTER** the binary is successfully saved to the store. That makes sense.